### PR TITLE
[9.x.x] Obsoleting FrameCleanup(), Adding OnCameraSetup() and OnCameraCleanup()

### PR DESCRIPTION
**Pull request by ellioman**
### Purpose of this PR
**This PR does four things:**
1. Adds ScriptableRenderPass.OnCameraSetup()
2. Makes the ScriptableRenderPass.FrameCleanup() obsolete
3. Adds ScriptableRenderPass.OnCameraCleanup() as its replacement.
4. Updates the render passes URP uses to the new naming

**There are two reasons for this PR:**
**First**, there's an issue in VR with how the render passes are getting temporary render textures and releasing them. Currently they are being created in ScriptableRenderPass.Configure() function and released in ScriptableRenderPass.FrameCleanup(). The problem is that Configure() is called **once per eye for each camera** but FrameCleanup() **once for each camera.** ScriptableRenderPass therefore needed a function that gets called once per camera before calling Configure().

**Second,** the FrameCleanup() is an incorrect name for the function as it is called per camera. So it made sense to include the function name change here as we created OnCameraSetup().

---
### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---

### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs [hunk 1]
```diff
@@ -152,6 +152,20 @@ public void ConfigureClear(ClearFlag clearFlag, Color clearColor)
...
+        /// Override this method if you need to to configure render targets and their clear state, and to create temporary render target textures.
+        /// If a render pass doesn't override this method, this render pass renders to the active Camera's render target.
+        /// You should never call CommandBuffer.SetRenderTarget. Instead call <c>ConfigureTarget</c> and <c>ConfigureClear</c>.
+        /// </summary>
+        /// <param name="cmd">CommandBuffer to enqueue rendering commands. This will be executed by the pipeline.</param>
+        /// <param name="cameraTextureDescriptor">Render texture descriptor of the camera render target.</param>
+        /// <param name="renderingData">Current rendering state information</param>
+        /// <seealso cref="ConfigureTarget"/>
+        /// <seealso cref="ConfigureClear"/>
+        public virtual void FrameSetup(CommandBuffer cmd, RenderTextureDescriptor cameraTextureDescriptor, ref RenderingData renderingData)
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs]: I think in RenderingData.CameraData we already have cameraTargetDescriptor, so cameraTextureDescriptor is redundant if we have renderingData.

File: com.unity.render-pipelines.universal/CHANGELOG.md [hunk 2]
```diff
@@ -40,6 +41,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - Renderer Feature list is now redesigned to fit more closely to the Volume Profile UI, this vastly improves UX and reliability of the Renderer Features List.
 - Default color values for Lit and SimpleLit shaders changed to white due to issues with texture based workflows.
 - You can now subclass ForwardRenderer to create a custom renderer based on it.
+- DepthOnlyPass, CopyDepthPass and CopyColorPass now use FrameSetup() instead of Configure() to set up their passes before executing
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/CHANGELOG.md]: Maybe we should say why is this useful/ what issue we are fixing with this.

File: com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs (hunk 1 above)
Comment (ellioman) [reply com.unity.render-pipelines.universal/Runtime/Passes/ScriptableRenderPass.cs]: Correct! I have now removed it.

File: com.unity.render-pipelines.universal/CHANGELOG.md (hunk 2 above)
Comment (ellioman) [reply com.unity.render-pipelines.universal/CHANGELOG.md]: I have now updated the changelog. Please take another look.

**Pull request merged**